Lib/sqlite3/test/userfunctions.py

```diff
@@ -276,18 +277,24 @@ def CheckAnyArguments(self):
         val = cur.fetchone()[0]
         self.assertEqual(val, 2)
 
+    @unittest.skipIf(sqlite.sqlite_version_info < (3, 8, 3), "Requires SQLite 3.8.3 or higher")
     def CheckFuncNonDeterministic(self):
+        """Non-deterministic functions cannot be used with partial indices"""
         mock = unittest.mock.Mock(return_value=None)
-        self.con.create_function("deterministic", 0, mock, deterministic=False)
-        self.con.execute("select deterministic() = deterministic()")
-        self.assertEqual(mock.call_count, 2)
+        self.con.create_function("nondeterministic", 0, mock, deterministic=False)
+        with self.assertRaises(sqlite.OperationalError) as cm:
+            self.con.execute("create index t on test(t) where nondeterministic() is not null")
+        self.assertEqual(str(cm.exception), "non-deterministic functions prohibited in partial index WHERE clauses")
```

Comment:
I suggest to not match the whole error message, only check if it contains "deterministic", something like:

```suggestion
        # non-deterministic functions prohibited in partial index WHERE clauses
        self.assertIn("non-deterministic functions prohibited", str(cm.exception))
```

Reply:
I agree. I see from the CI that earlier versions of SQLite produces slightly different messages. I was thinking something like `self.assertOk(str(cm.exception).contains("prohibited"))`.

Reply:
Also, I see from the CI that the SQLite version on the Ubuntu machine in the Azure pipelines test actually disallows _any_ function in partial index clauses, so we need another way of asserting this. I tried with `CREATE TABLE test(t TEXT CHECK (nondeterministic())`, but I was unable to trigger an SQLite "exception". I'll do some more experiments in a few hours.

Reply:
Well, if tests still fail with self.assertIn("non-deterministic functions prohibited", str(cm.exception)), I suggest to even remove the test on the error message. Expecting an exception is already good enough.

Reply:
Sorry, for being unclear. The Azure pipeline test failed both at the deterministic and the non-deterministic test, but for different reasons:
- The deterministic test failed because for this specific SQLite version, even deterministic functions were disallowed for the given SQL expression.
- The non-deterministic test failed because the expected error message was slightly different.

Reply:
Another possibility would be to require a newer SQLite version. After all, it's been a while since 3.7.3 was released...

Reply:
This still fails. I've fixed `CheckFuncNonDeterministic` locally but not pushed it yet. FYI:
```
-        self.assertIn("non-deterministic functions prohibited", str(cm.exception))
+        self.assertIn("prohibited", str(cm.exception))
```

However, `CheckFuncDeterministic` still fails, as I described in an earlier comment. I'll see if I can find a workaround. I won't push anything until the `CheckFuncDeterministic` issue is resolved. No need to run the CI again when we know it will fail.

Reply:
Resolved in b969ae5a60d984dde146e01e903b9d0bacf909a7 
Tested with sqlite 3.32.1, 3.15.0, 3.14.2 and 3.11.0.